ably/rest_client_test.go

```diff
@@ -270,11 +270,56 @@ func TestRSC7(t *testing.T) {
 			t.Errorf("expected %s got %s", ably.AblyVersion, h)
 		}
 	})
-	t.Run("must set lib header", func(t *testing.T) {
-		h := req.Header.Get(ably.AblyLibHeader)
-		if h != ably.LibraryString {
-			t.Errorf("expected %s got %s", ably.LibraryString, h)
+}
+
+func TestRest_RSC7_AblyAgent(t *testing.T) {
+	t.Parallel()
+
+	app, err := ablytest.NewSandbox(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer app.Close()
+
+	t.Run("RSC7d2 : Should set fallback host header", func(t *testing.T) {
+		t.Parallel()
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusInternalServerError)
+		}))
+		defer server.Close()
+
+		opts := []ably.ClientOption{
+			ably.WithEnvironment(ablytest.Environment),
+			ably.WithTLS(false),
+			ably.WithUseTokenAuth(true),
 		}
+		// set up the proxy to forward all requests except a specific fallback to the server,
+		// whilst that fallback goes to the regular endpoint
```

Comment:
What does this comment have to do with testing the Ably-Agent header?

Reply:
Yes, misleading comment